### PR TITLE
Make tests more resilient to service latency

### PR DESCRIPTION
**Pull request by drwill-ms**
A recent PR had to run the pipelines 5 times because some tests that use this flow kept failing after retries. But the retry settings, at least with the latest retry policy, are too meager. The tests didn't run for more than a few seconds, so this should give them more time in the event that the device registry is running slow.

### CODE REVIEW

**Review by patilsnr — APPROVED** [review]

**Review by timtay-microsoft — APPROVED** [review]

File: e2e/test/helpers/TestDevice.cs [hunk 1]
```diff
@@ -26,12 +26,12 @@ public enum ConnectionStringAuthScope
...
     public class TestDevice : IDisposable
     {
-        private const int MaxRetryCount = 5;
+        private const int MaxRetryCount = 20;
         private static readonly HashSet<IotHubServiceErrorCode> s_throttlingStatusCodes = new() { IotHubServiceErrorCode.ThrottlingException };
         private static readonly HashSet<IotHubServiceErrorCode> s_retryableStatusCodes = new(s_throttlingStatusCodes) { IotHubServiceErrorCode.DeviceNotFound };
         private static readonly SemaphoreSlim s_semaphore = new(1, 1);
 
-        private static readonly IRetryPolicy s_retryPolicy = new ExponentialBackoffRetryPolicy(MaxRetryCount, TimeSpan.FromSeconds(10));
+        private static readonly IRetryPolicy s_retryPolicy = new IncrementalDelayRetryPolicy(MaxRetryCount, TimeSpan.FromSeconds(1), TimeSpan.FromSeconds(3));
```
Comment (drwill-ms) [inline e2e/test/helpers/TestDevice.cs:34]: Same change here.

**Review by abhipsaMisra — APPROVED** [review]

File: e2e/test/helpers/TestDevice.cs [hunk 2]
```diff
@@ -121,7 +121,8 @@ await RetryOperationHelper
                             throw new IotHubServiceException(
                                 $"Created device {requestDevice.Id} not yet gettable from IoT hub.",
                                 HttpStatusCode.NotFound,
-                                IotHubServiceErrorCode.DeviceNotFound);
+                                // Using this causes the exception to be retriable, even though a better match would be DeviceNotFound.
```
Comment (drwill-ms) [inline e2e/test/helpers/TestDevice.cs:124]: remap to make retryable.

**Review by brycewang-microsoft — APPROVED** [review]

**Review by tmahmood-microsoft — APPROVED** [review]

**Review by brycewang-microsoft — APPROVED** [review]
Glad to see that we are covering more cases for the errors returned by service!

**Pull request merged**